### PR TITLE
Add transitionDuration property to app-drawer

### PR DESCRIPTION
**Pull request by keanulee**
This allows you to disable transitions (by setting `transitionDuration` to 0). This is a property instead of a CSS custom property because we need to read its value to see if it's 0 - if it is, there won't be a `transitionend` event and we call the `_resetDrawerState` method directly. (An alternative would have been to use `getComputedStyle()`, but that would be expensive.)

cc @frankiefu @blasten

### CODE REVIEW

File: app-drawer/app-drawer.html [hunk 1]
```diff
@@ -91,9 +91,10 @@
         width: var(--app-drawer-width, 256px);
         padding: 120px 0;
 
-        transition: 0.2s ease;
         transition-property: -webkit-transform;
         transition-property: transform;
+        transition-duration: inherit;
+        transition-timing-function: inherit;
```
Comment (blasten) [inline app-drawer/app-drawer.html]: yeah, it doesn't work in Safari, but it does work in Chrome and Firefox. http://jsbin.com/fifewi/edit?html,console.

**Comment by frankiefu** [comment]
FWIW, the prefix webkit (-webkit-transition-duration: inherit;) seems to work: http://jsbin.com/falekapese/edit?html,console
Also, unprefix seems to be fixed in webkit nightly.

**Comment by keanulee** [comment]
@frankiefu PTAL

**Comment by frankiefu** [comment]
LGTM

**Pull request merged**